go-controller/pkg/ovn/gateway_init.go

```diff
@@ -521,7 +521,7 @@ func (oc *DefaultNetworkController) addExternalSwitch(prefix, interfaceID, nodeN
 		},
 		Name: interfaceID,
 	}
-	if vlanID != nil {
```

Comment:
I was speaking to @jcaamano about the upgrades part and then he rightfully pointed out that we are creating new model for the switch port without the tag request, so existing switch ports's tag requests will get auto updated

so no need for any additional logic. I'll make sure QE tests this either ways and update you back here.

Reply:
tested it and it works